app/scripts/auto-type/emitter/auto-type-emitter-linux.js

```diff
@@ -61,13 +61,11 @@ const ModMap = {
 
 const AutoTypeEmitter = function (callback, windowId) {
     this.callback = callback;
-    if (typeof windowId !== 'undefined' && windowId) {
-        this.windowParameter = '--window ' + windowId + ' ';
```

Comment:
I'm thinking, does it break if we leave `windowParameter` as well? For example, if a window failed to activate for some reason, or if another window was opened during auto-type, keystrokes may be sent to another application. In theory it should work with both, but if it doesn't, let's remove it for sure.

Reply:
I've been running the code for the past 2 days and it's been solid for me.

In all my tests (via app and xdotool CLI), I just can't get `xdotool` to do the right when specifying the window. For example, in a terminal, I can run `xdotool type --window <terminals window_id> "hello"` (targeting the window I'm already in) and nothing happens. It only works if I remove that parameter. That suggests to me that leaving it off is the only way it works.

The key appears to be in the "SENDEVENT NOTES" section of the xdotool manpage:

> If you are trying to send key input to a specific window, and it does not
> appear to be working, then it's likely your application is ignoring the events
> xdotool is generating.  This is fairly common.
> 
> Sending keystrokes to a specific window uses a different API than simply typing
> to the active window. If you specify 'xdotool type --window 12345 hello'
> xdotool will generate key events and send them directly to window 12345.
> However, X11 servers will set a special flag on all events generated in this
> way (see XEvent.xany.send_event in X11's manual). Many programs observe this
> flag and reject these events.
> 
> It is important to note that for key and mouse events, we only use XSendEvent
> when a specific window is targeted. Otherwise, we use XTEST.
> 
> Some programs can be configured to accept events even if they are generated by
> xdotool. Seek the documentation of your application for help.
> 
> Specific application notes (from the author's testing): * Firefox 3 seems to
> ignore all input when it does not have focus.  * xterm can be configured while
> running with ctrl+leftclick, 'Allow SendEvents' * gnome-terminal appears to
> accept generated input by default.

That suggest that the use of the `--window` parameter using the XsendEvent will not work for all apps, while the XTEST does appear to work universally. To me that suggests that leaving the option out is the better choice.

Using both via config may also be doable, but might just lead to user confusion and frustration.

Reply:
Makes sense, thanks a lot for digging that far! I don't think it needs to be configurable since it seems to be broken for you.